clients/src/main/java/org/apache/kafka/clients/InFlightRequests.java

```diff
@@ -32,8 +31,7 @@
 
     private final int maxInFlightRequestsPerConnection;
     private final Map<String, Deque<NetworkClient.InFlightRequest>> requests = new HashMap<>();
-    /** Thread safe total number of in flight requests. */
-    private final AtomicInteger inFlightRequestCount = new AtomicInteger(0);
+    private int inFlightRequestCount = 0;
```

Comment:
This is intentionally thread-safe as per the documentation, so it's not safe to change it.

Reply:
Yes.  I saw the comment.  I'm not sure what the rational is. though. Multiple threads can access the same native int OK.  Can make it volatile if that is preferable.  Is it just that reading can take place multi-threaded?  Clearly the increment/decrement are not multi-threaded operations

Reply:
Yes, reading can be done from multiple threads. `volatile` would probably be enough for this use case.

Reply:
OK.  Changed to volatile.  This seems optimal to me in terms of performance and clarity.